### ISSUE TITLE
20.11 doesn't build with MIRA OFF

### ISSUE THREAD

**Issue by Jolly-Pirate** [steemit/steem#3369]
Building a witness node with the retagged 20.11 https://github.com/steemit/steem/tree/v0.20.11 fails with `ENABLE_MIRA=OFF`, but succeeds with `ENABLE_MIRA=ON`. It seems it's trying to test MIRA even though it's turned off!?

After a flood of red errors, I get this:
```
/root/steem/libraries/chainbase/include/chainbase/chainbase.hpp:1146:28: error: 'const ObjectType& chainbase::database::create(Constructor&&) [with ObjectType = test_object2; Constructor = basic_test(const std::vector<long unsigned int>&, std::function<void(Object&)>, chainbase::database&) [with IndexType = mira::multi_index_adapter<test_object2, boost::multi_index::indexed_by<mira::multi_index::ordered_unique<boost::multi_index::tag<ordered_idx2, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na>, mira::multi_index::member<test_object2, chainbase::oid<test_object2>, &test_object2::id> >, mira::multi_index::ordered_unique<boost::multi_index::tag<composite_ordered_idx2, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na>, mira::multi_index::composite_key<test_object2, mira::multi_index::member<test_object2, unsigned int, &test_object2::val>, mira::multi_index::member<test_object2, chainbase::oid<test_object2>, &test_object2::id> > > >, boost::interprocess::allocator<test_object2, boost::interprocess::segment_manager<char, boost::interprocess::rbtree_best_fit<boost::interprocess::mutex_family>, boost::interprocess::iset_index> > >; Object = test_object2; Order = ordered_idx2]::<lambda(test_object2&)>]', declared using local type 'basic_test(const std::vector<long unsigned int>&, std::function<void(Object&)>, chainbase::database&) [with IndexType = mira::multi_index_adapter<test_object2, boost::multi_index::indexed_by<mira::multi_index::ordered_unique<boost::multi_index::tag<ordered_idx2, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na>, mira::multi_index::member<test_object2, chainbase::oid<test_object2>, &test_object2::id> >, mira::multi_index::ordered_unique<boost::multi_index::tag<composite_ordered_idx2, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na>, mira::multi_index::composite_key<test_object2, mira::multi_index::member<test_object2, unsigned int, &test_object2::val>, mira::multi_index::member<test_object2, chainbase::oid<test_object2>, &test_object2::id> > > >, boost::interprocess::allocator<test_object2, boost::interprocess::segment_manager<char, boost::interprocess::rbtree_best_fit<boost::interprocess::mutex_family>, boost::interprocess::iset_index> > >; Object = test_object2; Order = ordered_idx2]::<lambda(test_object2&)>', is used but never defined [-fpermissive]
          const ObjectType& create( Constructor&& con )
                            ^~~~~~
libraries/mira/test/CMakeFiles/mira_test.dir/build.make:62: recipe for target 'libraries/mira/test/CMakeFiles/mira_test.dir/test.cpp.o' failed
make[2]: *** [libraries/mira/test/CMakeFiles/mira_test.dir/test.cpp.o] Error 1
CMakeFiles/Makefile2:1622: recipe for target 'libraries/mira/test/CMakeFiles/mira_test.dir/all' failed
make[1]: *** [libraries/mira/test/CMakeFiles/mira_test.dir/all] Error 2
Makefile:129: recipe for target 'all' failed
make: *** [all] Error 2
```

**Comment by mvandeberg** [comment]
Duplicate of #3349

The retagged 20.11 just added the openssl fix.

**Comment by Jolly-Pirate** [comment]
@mvandeberg it's not the openssl bug, it's the MIRA OFF problem, the fix is not in the retag!
https://github.com/steemit/steem/blob/v0.20.11/libraries/mira/test/CMakeLists.txt

**Comment by mvandeberg** [comment]
Yes, I know it is not in the retag. I say the retag *just* added the openssl fix. As in *only* added that fix.

0.20.11 is MIRA. It is the same as 0.20.10. Apply the openssl fix to 0.20.10 and you can still run an up to date node.

**Comment by Jolly-Pirate** [comment]
Oh I understood that "just" meant recently. "Only" is a clearer wording.

**Comment by Jolly-Pirate** [comment]
The openssl patch 8e2e0f774332b9e6a2d3942ceb3e707bcc5867b2 doesn't fix v0.20.10 on ubuntu 18.04 + openssl 1.1.1.

**Comment by mvandeberg** [comment]
I confirmed on a fresh Ubuntu 18.04 image that explicitly building `steemd` and `cli_wallet` targets works on `v0.20.11` with `ENABLE_MIRA=OFF`.

Building `make all` with `ENABLE_MIRA=OFF` works on `master` and will be in the next release.
